recirq/quantum_chess/quantum_board.py

```diff
@@ -598,6 +689,7 @@ def do_move(self, m: move.Move) -> int:
 
         # Add move to move_history
         self.move_history.append(m)
+        self.move_history_probabilities_cache.append((-1, None))
```

Comment:
move_history_probabilities_cache entry should be removed in undo_move

Reply:
undo_move calls `with_state()`, which clears move_history_probabilities_cache and move_history

Reply:
OK, then LGTM!